test/Profiler/coverage_optimized_report.swift

```diff
@@ -14,7 +14,7 @@
 // REQUIRES: executable_test
 // REQUIRES: OS=macosx
 
-// CHECK: "lines":{"count":9,"covered":6{{.*}}"functions":{"count":5,"covered":3
+// CHECK: "lines":{"count":9,"covered":5{{.*}}"functions":{"count":5,"covered":3
```

Comment:
What line isn't covered now?

Reply:
It's the closing `}` of `if bar()`, previously we emitted a region for after the `if`, but we no longer do so if the counter hasn't changed from the parent. That means in this case that we'll no longer consider that line to be covered since it will now use the counter for the body of the `if`, but I think that's reasonable enough.